### ISSUE TITLE
ImportError: cannot import name 'RequestsTransport' from 'azure.core.pipeline.transport'

### ISSUE THREAD

**Issue by aabuhasna** [Azure/azure-sdk-for-python#29245]
- **Package Name: azure-storage-queue**: 
- **Package Version: latest**: 
- **Operating System: Windows 11**:
- **Python Version: 3.8.10**: 

**Describe the bug**
when trying to create a new queue it failed to import QueueClient from the Azure storage queue module (library)

**To Reproduce**
1.  Follow the same procedure inside https://learn.microsoft.com/en-us/azure/storage/queues/storage-python-how-to-use-queue-storage?tabs=python%2Cenvironment-variable-windows

**Expected behavior**
Create a queue

**Screenshots**
![image](https://user-images.githubusercontent.com/114708215/223999284-b0d0148d-c403-4865-9be1-6b48dfe6c585.png)

**Additional context**

**Comment by kashifkhan** [comment]
Thank you for the feedback @aabuhasna . We will investigate and get back to you asap.

**Comment by vincenttran-msft** [comment]
Hi @aabuhasna thanks for your inquiry. I have just gone ahead and tried the setup process of `azure-storage-queue` using the latest package version and your Python version and was able to successfully run my code with no issues. With that being said, it seems like this is likely a configuration issue on your end.

Can you please verify that you have `azure-core` installed? You can verify this by executing `pip freeze` and ensuring you have an `azure-core` entry i.e. `azure-core==1.26.3`. If you are unsure, you can also share the results of `pip freeze` here.
Another good idea may be to reinstall any of the components necessary to avoid using anything that is out of date (i.e. Python, `azure-storage-queue`, `azure-core`, etc.)

With that being said, hopefully this helps resolve your issue, but I can verify that this is not an SDK issue and is likely a configuration issue or something else not related to the SDK.

Thanks!

**Comment by m-he** [comment]
I got similar error when create a python Azure Function, with python3.10, platform Linux, `azure-core==1.26.3`, `azure-storage-file-share===12.11.1` 
```
Microsoft.Azure.WebJobs.Host.FunctionInvocationException : Exception while executing function: Functions.myapp ---> Microsoft.Azure.WebJobs.Script.Workers.Rpc.RpcException : Result: Failure
Exception: ImportError: requests package is not installed. Please check the requirements.txt file for the missing module. For more info, please refer the troubleshooting guide: https://aka.ms/functions-modulenotfound
Stack: File '/azure-functions-host/workers/python/3.10/LINUX/X64/azure_functions_worker/dispatcher.py', line 357, in _handle__function_load_request
func = loader.load_function(
File '/azure-functions-host/workers/python/3.10/LINUX/X64/azure_functions_worker/utils/wrappers.py', line 48, in call
raise extend_exception_message(e, message)
File '/azure-functions-host/workers/python/3.10/LINUX/X64/azure_functions_worker/utils/wrappers.py', line 44, in call
return func(*args, **kwargs)
File '/azure-functions-host/workers/python/3.10/LINUX/X64/azure_functions_worker/loader.py', line 132, in load_function
mod = importlib.import_module(fullmodname)
File '/usr/local/lib/python3.10/importlib/__init__.py', line 126, in import_module
return _bootstrap._gcd_import(name[level:], package, level)
File '<frozen importlib._bootstrap>', line 1050, in _gcd_import
File '<frozen importlib._bootstrap>', line 1027, in _find_and_load
File '<frozen importlib._bootstrap>', line 1006, in _find_and_load_unlocked
File '<frozen importlib._bootstrap>', line 688, in _load_unlocked
File '<frozen importlib._bootstrap_external>', line 883, in exec_module
File '<frozen importlib._bootstrap>', line 241, in _call_with_frames_removed
File '/home/site/wwwroot/myapp/__init__.py', line 10, in <module>
from azure.storage.fileshare import ShareClient, StorageErrorCode
File '/home/site/wwwroot/.python_packages/lib/site-packages/azure/storage/fileshare/__init__.py', line 9, in <module>
from ._file_client import ShareFileClient
File '/home/site/wwwroot/.python_packages/lib/site-packages/azure/storage/fileshare/_file_client.py', line 25, in <module>
from ._shared.base_client import StorageAccountHostsMixin, parse_connection_str, parse_query
File '/home/site/wwwroot/.python_packages/lib/site-packages/azure/storage/fileshare/_shared/base_client.py', line 27, in <module>
from azure.core.pipeline.transport import RequestsTransport, HttpTransport
File '/home/site/wwwroot/.python_packages/lib/site-packages/azure/core/pipeline/transport/__init__.py', line 77, in __getattr__
raise ImportError('requests package is not installed')
```

**Comment by aabuhasna** [comment]
Hi @vincenttran-msft thank you for your feedback, I am using azure-core version 1.26.3 combined with azure-storage-queue version 12.6.0 after spending several hours troubleshooting what I have done wrong with my script I decided to go with azure-storage-queue 12.5.0 for testing and it worked like a charm.

In order to go live with my code with the latest version, I will try to use a new machine (using the same version of MS Windows 11 22H2)  and install all the component there, I will keep you updated with the result of my testing.

**Comment by aabuhasna** [comment]
Hi @vincenttran-msft sorry about the delay, I have tested the script again using the latest version of Azure storage queue and it was successful. It seems the issue was with my machine. sorry about the inconvenience.

**Comment by b-gonzalez** [comment]
I am running into this same issue. Let me know if you'd like me to create a new issue. 

I am trying to follow this tutorial:

https://learn.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-python?tabs=managed-identity%2Croles-azure-portal%2Csign-in-azure-cli

When I get to this line of code:

`from azure.storage.blob import BlobServiceClient, BlobClient, ContainerClient`

I get the following error [ImportError: cannot import name 'RequestsTransport' from 'azure.core.pipeline.transport']

I can import and reference the azure core pipeline transport e.g.

`import azure.core.pipeline.transport as acpt`

but if I try `from azure.core.pipeline.transport import RequestsTransport` I get the same error

The version of azure-core I had installed was 1.26.4. I tried downgrading to 1.26.3 but it made no difference.
